### PR TITLE
Add Landing component to Home view

### PR DESCRIPTION
**Pull request by w-gao**
Problem
=======
Landing component [Jira](https://cruzhacks-dev.atlassian.net/browse/C2D-27).

Solution
========
What I/we did to solve this problem
* Add Landing component to Home view

Change Summary:
---------------
* Add Landing component to Home view
* Modified the Button component to work better with external links
* Also added `modifier` prop to the Button component so we don't have the pass Button-specific class names to use its modifiers (e.g.: `className="landing-component__Button" modifier="primary"` instead of `className="landing-component__Button Button--primary"`)

Dev-Ops
=======
If you added an ENV variable, please check off that you've updated the following  
- [ ] Key & Sample value to `.env` & `sample.env` 
- [ ] GCP Secret Manager (Both development and production)
- [ ] Github Secrets (Added a development and production variable)
- [ ] Github Workflows `.github/workflows/dev.yml` & `.github/workflows/prod.yml`
- [ ] webpack-config.js

Images/Important Notes (optional):
-----------------------

### CODE REVIEW

**Comment by w-gao** [comment]
Styling needs more work. Also have to add the background graphics.

**Comment by w-gao** [comment]
The Email subscription component (#41) needs to be added in, but this should mostly be ready now.

**Review by krishnakpandian — COMMENTED** [review]
This looks amazing!! Let's wait until we merge the subscription component before merging

File: src/views/Home/components/Landing/index.tsx
Comment (w-gao) [inline src/views/Home/components/Landing/index.tsx:54]: Is this the link we want for this button?

**Review by w-gao — COMMENTED** [review]
I modified the Button component (see PR description for details) that needs a re-review. I can make that into a separate PR if that needs more discussion.

File: src/views/Home/components/Landing/index.scss
Comment (krishnakpandian) [inline src/views/Home/components/Landing/index.scss:52]: On small-laptop, I think having a margin-top of 0 seems to be a bit more natural but that's only my opinion on it, we can double check with design.

**Review by krishnakpandian — APPROVED** [review]
landing-input-box-sizing
This looks amazing overall but I feel like the height of the landing-text-box differs a bit from the figma. Again I think it's fine for the most part but it might be something worth looking into.

**Review by krishnakpandian — APPROVED** [review]
LGTM

File: src/views/Home/components/Landing/index.tsx
Comment (krishnakpandian) [reply src/views/Home/components/Landing/index.tsx:54]: This is what we have tentatively linked publicly for 2022, so this is good.

**Pull request merged**
